lib/statsd/instrument.rb

```diff
@@ -277,17 +279,25 @@ def backend
   #      http_response = StatsD.measure('HTTP.call.duration') do
   #        HTTP.get(url)
   #      end
-  def measure(key, value = nil, *metric_options, &block)
+  def measure(key, value = nil, *metric_options)
     if value.is_a?(Hash) && metric_options.empty?
       metric_options = [value]
       value = value.fetch(:value, nil)
     end
 
-    result = nil
-    value  = 1000 * StatsD::Instrument.duration { result = block.call } if block_given?
-    metric = collect_metric(hash_argument(metric_options).merge(type: :ms, name: key, value: value))
-    result = metric unless block_given?
-    result
+    if block_given?
+      result = nil
+      start = StatsD::Instrument.current_timestamp
+      begin
+        result = yield
+      ensure
+        value = 1000 * StatsD::Instrument.current_timestamp - start
```

Comment:
Can you add parentheses here? `1000 * (StatsD::Instrument.current_timestamp - start). I am not too sure what the rules are in Ruby, with parentheses it's clear that it will be correct.